leaderelection/config.go

```diff
@@ -34,13 +34,17 @@ import (
 
 const (
 	configMapNameEnv = "CONFIG_LEADERELECTION_NAME"
-	// KnativeResourceLock is the only supported lock mechanism for Knative.
-	KnativeResourceLock = resourcelock.LeasesResourceLock
+	// defaultKnativeResourceLock is the default lock mechanism for Knative.
+	defaultKnativeResourceLock = resourcelock.LeasesResourceLock
 )
 
-// MaxBuckets is the maximum number of buckets to allow users to define.
-// This is a variable so that it may be customized in the binary entrypoint.
-var MaxBuckets uint32 = 10
+var (
+	// MaxBuckets is the maximum number of buckets to allow users to define.
+	// This is a variable so that it may be customized in the binary entrypoint.
+	MaxBuckets uint32 = 10
+	// supportedResourceLocks is the supported resource lock types for Knative.
+	supportedResourceLocks = sets.NewString(defaultKnativeResourceLock, resourcelock.EndpointsResourceLock)
```

Comment:
Do we need this? This is mot likely going to be used internally?

Reply:
removed